### ISSUE TITLE
infer break point after phasing

### ISSUE THREAD

**Issue by Yiguan** [whatshap/whatshap#295]
Hi,

`whatshap` has phased my vcf files, but I am wondering how to infer break points(crossover or gene conversion) based on the phased haplotypes. 

I found `duoHMM` [here](https://mathgen.stats.ox.ac.uk/genetics_software/duohmm/duohmm.html#home) could infer the probability of break points. The output from `duoHMM` is, for example:

```
CHILD      PARENT     START    END      PROB_RECOMBINATION
sample_74  sample_32  804515   825672   0.100080297932
sample_95  sample_32  1307248  1316440  0.139441220727
sample_84  sample_32  1307248  1316440  0.105295050636
```
But `duoHMM` seems to work only with `shapeit`, and it cannot be used with `whatshap`.

In short, does `whatshap` support the inference of break points?

Thanks in advance!

**Comment by tobiasmarschall** [comment]
Whatshap cannot infer recombination rates across a population. That'd require to work on many samples at once and that's not how WhatsHap operates. What it can do (not sure that's relevant to you) is call meiotic recombination sites when doing pedigree-based phasing (e.g. in a trio).

**Comment by Yiguan** [comment]
> Whatshap cannot infer recombination rates across a population. That'd require to work on many samples at once and that's not how WhatsHap operates. What it can do (not sure that's relevant to you) is call meiotic recombination sites when doing pedigree-based phasing (e.g. in a trio).

My data is trio data(father - mother - children), I have used `whatshap` to phase the data. But it doesn't tell the meiotic recombination. I guess I may be able to find the recombination if I manually check these phased haplotypes.  But that's not preferred. Ideally, `whatshap` could automatically output where the recombination occurs in the trio data.

**Comment by tobiasmarschall** [comment]
There is an option `--recombination-list` that you can use to output recombination events. It tends to contain false positives in regions of segmental duplications (where input read alignments and/or genotypes can be wrong), but it should do a reasonable job in terms of sensitivity. You will want to filter it for predicted events that are close together...

**Comment by Yiguan** [comment]
> There is an option `--recombination-list` that you can use to output recombination events. It tends to contain false positives in regions of segmental duplications (where input read alignments and/or genotypes can be wrong), but it should do a reasonable job in terms of sensitivity. You will want to filter it for predicted events that are close together...

Thank you. I tried `--recombination-list` and it did output a list of possible recombination events. But as you said, it has many false positives, and I am trying to figure out the filtering...

**Comment by tobiasmarschall** [comment]
As I said above, the distance between the reported events should be a good indication. Maybe you want to start by plotting the inheritance state as reported in this list along a chromosome. That will likely show you that there are places where there's a lot of switching due to mismapping (not trustworthy), but hopefully also more long range switches, i.e. where the inheritance switches and then stays this way.

**Comment by Yiguan** [comment]
> As I said above, the distance between the reported events should be a good indication. Maybe you want to start by plotting the inheritance state as reported in this list along a chromosome. That will likely show you that there are places where there's a lot of switching due to mismapping (not trustworthy), but hopefully also more long range switches, i.e. where the inheritance switches and then stays this way.

I have visualised the recombination events along the chromosome 4 (chromosome length ~1.3Mb in *Drosophila* ).

![the recombination plot](https://github.com/Yiguan/inferRecombination/blob/main/recombinationCost.png)

The plot includes 5 children from the same parents in a *Drosophila* family. The coloured bar indicates the region where `whatshap` suggested a breakpoint. 

As I don't have the genetic map for *Drosophila*, I used a uniform recombination rate (from 1cM/Mb to 3cM/Mb, showed in different colours in the plot). The plot seems to suggest that the recombination rate doesn't affect the results except a bit difference in recombination cost(Y-axis). I am happy with this result since I don't need to worry about the recombination rate as a parameter in `whatshap`.

However, `whatshap` reported too many recombinations, particularly in `sample75`, which is very unlikely to happen in such a short chromosome. The recombination costs, phred-scaled probability that a recombination event occurs if I understand correctly, are very high (>30). 

Let check some SNP markers.

Some seem perfect, for example in sample74.
![sample74](https://github.com/Yiguan/inferRecombination/blob/main/sample_74_2.png): purple shaded region indicates the recombination region. NOTE: X-axis is the SNP marker, not bp.

```
child_id   position1  position2  transmitted_hap_father1  transmitted_hap_father2  transmitted_hap_mother1  transmitted_hap_mother2  recombination_cost
sample_74  1281401    1286595    1                        0                        0                        0                        39
sample_74  1305846    1305878    0                        0                        0                        1                        61
sample_74  1307028    1307100    0                        0                        1                        0                        58
sample_74  1307565    1316261    0                        0                        0                        1                        37
sample_74  1318317    1318730    0                        0                        1                        0                        50
```
The SNP markers perfectly match the inferred breakpoints: the first recombination occurred in the father and the other four recombinations occurred in the mother.

But some results seem confusing, for example in sample62

![sample62](https://github.com/Yiguan/inferRecombination/blob/main/sample_62_1.png)

```
child_id   position1  position2  transmitted_hap_father1  transmitted_hap_father2  transmitted_hap_mother1  transmitted_hap_mother2  recombination_cost
sample_62  1281401    1286595    1                        0                        0                        0                        39
sample_62  1305923    1306074    0                        0                        0                        1                        55
sample_62  1306188    1306395    0                        0                        1                        0                        53
sample_62  1307565    1316261    0                        0                        0                        1                        37
sample_62  1318246    1318286    0                        0                        1                        0                        60
```

Based on the SNP markers in the plot, there are no need to have recombinations in the mother(the second and third purple shaded regions in the plot). 

Please let me know if somewhere I doesn't make it clear. 
Looking forward your suggestion!
Thanks so much!
